### PR TITLE
HAWKULAR-235 Make Pinger listen to URLs removed

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@pilhuhn could you please review, once the CI passes?

File: modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java [hunk 1]
```diff
@@ -119,20 +203,23 @@ public void call(Resource r) {
     @javax.annotation.Resource(lookup = "java:global/Hawkular/ObservableInventory")
     private Inventory.Mixin.Observable inventory;
 
-    final NewUrlsCollector newUrlsCollector = new NewUrlsCollector();
+    final UrlChangesCollector newUrlsCollector = new UrlChangesCollector();
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: variable naming?

File: modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java [hunk 2]
```diff
@@ -52,47 +55,128 @@
...
         }
+    }
 
+    /**
+     * Collects URL additions and removals reported by {@link PingManager#inventory} and synchronizes the various
+     * threads reporting the new URLs and those ones consuming them.
+     *
+     * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+     */
+    static class UrlChangesCollector {
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: Should that go into its own class file to not "pollute" this manager class? Probably along with the `UrlChange`

File: modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java (hunk 1 above)
Comment (ppalaga) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: +1

File: modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java (hunk 2 above)
Comment (ppalaga) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java]: +1, even unit testing will be cleaner

File: modules/pinger/src/main/java/org/hawkular/component/pinger/UrlChangesCollector.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+/**
+ * Collects URL additions and removals reported by {@link PingManager#inventory} and synchronizes the various threads
+ * reporting the new URLs and those ones consuming them.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class UrlChangesCollector {
+    /**
+     * An an add or remove operation that can be performed on a {@link Set}.
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/UrlChangesCollector.java]: Typo "An an"

File: modules/pinger/src/main/java/org/hawkular/component/pinger/UrlChangesCollector.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                    changes.add(new UrlChange(Action.created(), PingDestination.from(r)));
+                }
+            }
+        }
+    };
+
+    /**
+     * A callback for the {@link Inventory} that collects URL removals.
+     */
+    private final Action1<Resource> urlDeletedAction = new Action1<Resource>() {
```
Comment (pilhuhn) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/UrlChangesCollector.java]: Is it possible to only use one action callback for both cases and then internally inside `call` differentiate on the incoming event?

Comment (ppalaga) [inline modules/pinger/src/main/java/org/hawkular/component/pinger/UrlChangesCollector.java]: No, `call` get only a `Resource` and one cannot tell whether it was added or removed.

**Pull request merged**